File: src/main/java/com/pedigree/model/Event.java
package com.pedigree.model;

import java.time.LocalDate;

public class Event {
    private String type; // e.g., BIRTH, DEATH, MARRIAGE
    private LocalDate date;
    private String place;

    public String getType() { return type; }
    public void setType(String type) { this.type = type; com.pedigree.util.DirtyFlag.setModified(); }
    public LocalDate getDate() { return date; }
    public void setDate(LocalDate date) { this.date = date; com.pedigree.util.DirtyFlag.setModified(); }
    public String getPlace() { return place; }
    public void setPlace(String place) { this.place = place; com.pedigree.util.DirtyFlag.setModified(); }
}

File: src/main/kotlin/com/family/tree/model/Event.kt
package com.family.tree.model

import com.family.tree.util.DirtyFlag
import java.time.LocalDate

class Event {
    var type: String? = null // e.g., BIRTH, DEATH, MARRIAGE
        set(value) { field = value; DirtyFlag.setModified() }
    var date: LocalDate? = null
        set(value) { field = value; DirtyFlag.setModified() }
    var place: String? = null
        set(value) { field = value; DirtyFlag.setModified() }
}
